### PR TITLE
[CP] Add some columns and filters to citizen project list admin

### PR DESCRIPTION
**Pull request by B-Galati**

### CODE REVIEW

File: src/Admin/CitizenProjectAdmin.php [hunk 1]
```diff
@@ -31,13 +35,15 @@ class CitizenProjectAdmin extends AbstractAdmin
     private $manager;
     private $citizenProjectMembershipRepository;
     private $cachedDatagrid;
+    private $citizenProjectRepository;
```
Comment (ottaviano) [inline src/Admin/CitizenProjectAdmin.php]: it's used?

File: src/Admin/CitizenProjectAdmin.php [hunk 2]
```diff
@@ -277,9 +292,84 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
...
+                    }
+
+                    /** @var ProxyQuery|QueryBuilder $qb */
+                    if (!in_array('creator', $qb->getAllAliases(), true)) {
+                        $qb->leftJoin(Adherent::class, 'creator', Join::WITH, sprintf('creator.uuid=%s.createdBy', $alias));
+                    }
+
+                    $qb->andWhere(
+                            'LOWER(creator.emailAddress)=:emailAddress '.
+                            "OR $alias.createdBy=:uuid"
```
Comment (ottaviano) [inline src/Admin/CitizenProjectAdmin.php]: it's really necessary to search an email (`EmailType` field) in uuid column?

File: src/Admin/CitizenProjectAdmin.php [hunk 3]
```diff
@@ -277,9 +292,84 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
...
+                        return;
+                    }
+
+                    /** @var ProxyQuery|QueryBuilder $qb */
+                    if (!in_array('creator', $qb->getAllAliases(), true)) {
+                        $qb->leftJoin(Adherent::class, 'creator', Join::WITH, sprintf('creator.uuid=%s.createdBy', $alias));
+                    }
+
+                    $qb->andWhere(
+                            'LOWER(creator.emailAddress)=:emailAddress '.
```
Comment (ottaviano) [inline src/Admin/CitizenProjectAdmin.php]: you can avoid `LOWER` function on email field and on `_ci` collate, it's already saved as lowercase string:
```
//MembershipRequest
public function setEmailAddress(string $emailAddress): void
    {
        $this->emailAddress = mb_strtolower($emailAddress);
    }
```

File: src/Admin/CitizenProjectAdmin.php [hunk 4]
```diff
@@ -277,9 +292,84 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
...
+                            'LOWER(creator.firstName) LIKE :firstName '.
+                            "OR $alias.createdBy=:uuid"
+                        )
+                        ->setParameter('firstName', sprintf('%%%s%%', mb_strtolower($value['value'])))
+                        ->setParameter('uuid', $value['value']);
+
+                    return true;
+                },
+            ])
+            ->add('creatorLastName', CallbackFilter::class, [
```
Comment (ottaviano) [inline src/Admin/CitizenProjectAdmin.php]: maybe do juste one filter fied `Createur` for search in last and first name?

File: src/Admin/CitizenProjectAdmin.php (hunk 1 above)
Comment (B-Galati) [reply src/Admin/CitizenProjectAdmin.php]: No thanks !

File: src/Admin/CitizenProjectAdmin.php (hunk 3 above)
Comment (B-Galati) [reply src/Admin/CitizenProjectAdmin.php]: Yea great :-)

File: src/Admin/CitizenProjectAdmin.php (hunk 4 above)
Comment (B-Galati) [reply src/Admin/CitizenProjectAdmin.php]: I think we must stick to the current layout with lastname and firstname in different filter

File: src/Admin/CitizenProjectAdmin.php (hunk 2 above)
Comment (B-Galati) [reply src/Admin/CitizenProjectAdmin.php]: It's the way it is done and similar other field so I stick to it

Comment (ottaviano) [reply src/Admin/CitizenProjectAdmin.php]: why? 😕 
the input[type=email] field allows you to insert only a valid email address and it makes no sense to chech UUID column with an email.

Comment (B-Galati) [reply src/Admin/CitizenProjectAdmin.php]: I misunderstood what was done in the other fields.. all fixed now. Thank you very much :-)

**Pull request merged**